pkg/pipeline/write/write_ipfix.go

```diff
@@ -0,0 +1,535 @@
+		}
+	}
+
+	err = templateSet.AddRecord(elements, templateID)
+	if err != nil {
+		ilog.WithError(err).Error("Failed in Add Record")
+		return 0, err
+	}
+	_, err = exporter.SendSet(templateSet)
+	if err != nil {
+		ilog.WithError(err).Error("Failed to send template record")
+		return 0, err
+	}
+
+	return templateID, nil
+}
+
+func (t *writeIpfix) sendDataRecord(record config.GenericMap) error {
+	// Create data set with 1 data record
+
+	elements := make([]entities.InfoElementWithValue, 0)
+
+	err := addElementToTemplate("ethernetType", makeByteFromUint16(uint16(record["Etype"].(uint32))), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("flowDirection", makeByteFromUint8(uint8(record["FlowDirection"].(int))), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("sourceMacAddress", net.HardwareAddr(record["SrcMac"].(string)), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("destinationMacAddress", net.HardwareAddr(record["DstMac"].(string)), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	if IPv6Type == record["Etype"].(uint32) {
+		err = addElementToTemplate("sourceIPv6Address", net.ParseIP(record["SrcAddr"].(string)), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+		err = addElementToTemplate("destinationIPv6Address", net.ParseIP(record["DstAddr"].(string)), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+		err = addElementToTemplate("nextHeaderIPv6", makeByteFromUint8(uint8(record["Proto"].(uint32))), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = addElementToTemplate("sourceIPv4Address", net.ParseIP(record["SrcAddr"].(string)), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+		err = addElementToTemplate("destinationIPv4Address", net.ParseIP(record["DstAddr"].(string)), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+		err = addElementToTemplate("protocolIdentifier", makeByteFromUint8(uint8(record["Proto"].(uint32))), &elements, registry.IANAEnterpriseID)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = addElementToTemplate("sourceTransportPort", makeByteFromUint16(uint16(record["SrcPort"].(uint32))), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("destinationTransportPort", makeByteFromUint16(uint16(record["DstPort"].(uint32))), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("octetDeltaCount", makeByteFromUint64(record["Bytes"].(uint64)), &elements, registry.IANAEnterpriseID)
+	if err != nil {
+		return err
+	}
+	err = addElementToTemplate("flowStartMilliseconds", makeByteFromUint64(uint64(record["TimeFlowStartMs"].(int64))), &elements, registry.IANAEnterpriseID)
```

Comment:
From decode_protobuf.go, 
```
func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
	if flow == nil {
		return config.GenericMap{}
	}
	out := config.GenericMap{
		"FlowDirection":   int(flow.Direction.Number()),
		"Bytes":           flow.Bytes,
		"SrcAddr":         ipToStr(flow.Network.GetSrcAddr()),
		"DstAddr":         ipToStr(flow.Network.GetDstAddr()),
		"SrcMac":          macToStr(flow.DataLink.GetSrcMac()),
		"DstMac":          macToStr(flow.DataLink.GetDstMac()),
		"SrcPort":         flow.Transport.GetSrcPort(),
		"DstPort":         flow.Transport.GetDstPort(),
		"Etype":           flow.EthProtocol,
		"Packets":         flow.Packets,
		"Duplicate":       flow.Duplicate,
		"Proto":           flow.Transport.GetProtocol(),
		"TimeFlowStartMs": flow.TimeFlowStart.AsTime().UnixMilli(),
		"TimeFlowEndMs":   flow.TimeFlowEnd.AsTime().UnixMilli(),
		"TimeReceived":    time.Now().Unix(),
		"Interface":       flow.Interface,
	}
	return out
}
```
Since, it's part of the standard data, I am taking it as it.

Reply:
How do you handle, in general, if one of your expected fields does not exist? Won't we get an exception?

Reply:
I have added a nil condition check now. That should handle the corner cases